packages/dropdowns/src/elements/Trigger/Trigger.tsx

```diff
@@ -45,19 +46,35 @@ const Trigger: React.FunctionComponent<ITriggerProps> = ({ children, refKey, ...
     // Destructuring the `ref` argument lets us share it with PopperJS
     const { ref: rootPropsRefCallback, ...rootProps } = getRootProps();
 
+    const listboxToggleProps = getToggleButtonProps({
+      ...rootProps,
+      // Trigger usages do not include an role
+      role: null,
+      // Trigger usages do not include an associated label
+      'aria-labelledby': undefined,
+      ...triggerProps,
+      ...(children as any).props
+    });
+
+    const menuToggleProps = {
+      ...listboxToggleProps,
+      'aria-haspopup': 'true',
+      'aria-controls': listboxToggleProps['aria-owns'],
+      'aria-owns': null
+    };
```

Comment:
All props should be passed through `getToggleButtonProps()` in some way before being applied to the element. I think this would be more clear if the conditional (listbox vs menu) logic is moved above here somehow.

Reply:
> All props should be passed through getToggleButtonProps() in some way before being applied to the element.

The props are passed through `getToggleButtonProps()`. The spread (line 60) points directly to a `getToggleButtonProps` invocation. 

<kbd>
<img width="488" alt="Screen Shot 2020-05-05 at 12 37 37 PM" src="https://user-images.githubusercontent.com/1811365/81108608-cc955500-8ecd-11ea-9a54-0458f61074db.png">
</kbd>

Reply:
> I think this would be more clear if the conditional (listbox vs menu) logic is moved above here somehow.

Could you clarify how your suggestion might look? Specifically, which bit of logic should be moved above? It would help me understand why what I have isn't clear.

Reply:
`aria-haspopup`, `aria-control`, and `aria-owns` are not being passed through `getToggleButtonProps()` in this example.

Reply:
`aria-haspopup`, `aria-control`, and `aria-owns` aren't passed through because we want Downshift to take care of listbox props.

We want to override these props when Downshift is used to render menu dropdowns. This is why I chose the names "listboxToggleProps" versus "menuToggleProps". Note, Downshift does not support menu dropdowns out of the box — only listbox dropdowns.